plutarch-api/src/Plutarch/Api.hs

```diff
@@ -0,0 +1,1275 @@
+  = PMinting (Term s (PDataRecord '["_0" ':= Value.PCurrencySymbol]))
+  | PSpending (Term s (PDataRecord '["_0" ':= PTxOutRef]))
+  | PRewarding (Term s (PDataRecord '["_0" ':= PStakingCredential]))
+  | PCertifying (Term s (PDataRecord '["_0" ':= PDCert]))
+  deriving stock
+    ( -- | @since 2.0.0
+      Generic
+    )
+  deriving anyclass
+    ( -- | @since 2.0.0
+      PlutusType
+    , -- | @since 2.0.0
+      PIsData
+    , -- | @since 2.0.0
+      PEq
+    , -- | @since 2.0.0
+      PShow
+    )
+
+-- | @since 2.0.0
+instance DerivePlutusType PScriptPurpose where
+  type DPTStrat _ = PlutusTypeData
+
+-- | @since 2.0.0
+instance PUnsafeLiftDecl PScriptPurpose where
+  type PLifted PScriptPurpose = Plutus.ScriptPurpose
+
+-- | @since 2.0.0
+deriving via
+  (DerivePConstantViaData Plutus.ScriptPurpose PScriptPurpose)
+  instance
+    PConstantDecl Plutus.ScriptPurpose
+
+-- | @since 2.0.0
+newtype PTxOut (s :: S)
+  = PTxOut
+      ( Term
+          s
+          ( PDataRecord
+              '[ "address" ':= PAddress
+               , "value" ':= Value.PValue 'AssocMap.Sorted 'Value.Positive
+               , "datum" ':= POutputDatum
+               , "referenceScript" ':= PMaybeData PScriptHash
+               ]
+          )
+      )
+  deriving stock
+    ( -- | @since 2.0.0
+      Generic
+    )
+  deriving anyclass
+    ( -- | @since 2.0.0
+      PlutusType
+    , -- | @since 2.0.0
+      PIsData
+    , -- | @since 2.0.0
+      PDataFields
+    , -- | @since 2.0.0
+      PEq
+    , -- | @since 2.0.0
+      PShow
+    )
+
+-- | @since 2.0.0
+instance DerivePlutusType PTxOut where
+  type DPTStrat _ = PlutusTypeData
+
+-- | @since 2.0.0
+instance PUnsafeLiftDecl PTxOut where
+  type PLifted PTxOut = Plutus.TxOut
+
+-- | @since 2.0.0
+deriving via
+  (DerivePConstantViaData Plutus.TxOut PTxOut)
+  instance
+    PConstantDecl Plutus.TxOut
+
+{- | An input of the pending transaction.
+
+@since 2.0.0
+-}
+newtype PTxInInfo (s :: S)
+  = PTxInInfo
+      ( Term
+          s
+          ( PDataRecord
+              '[ "outRef" ':= PTxOutRef
+               , "resolved" ':= PTxOut
+               ]
+          )
+      )
+  deriving stock
+    ( -- | @since 2.0.0
+      Generic
+    )
+  deriving anyclass
+    ( -- | @since 2.0.0
+      PlutusType
+    , -- | @since 2.0.0
+      PIsData
+    , -- | @since 2.0.0
+      PDataFields
+    , -- | @since 2.0.0
+      PEq
+    , -- | @since 2.0.0
+      PShow
+    )
+
+-- | @since 2.0.0
+instance DerivePlutusType PTxInInfo where
+  type DPTStrat _ = PlutusTypeData
+
+-- | @since 2.0.0
+instance PUnsafeLiftDecl PTxInInfo where
+  type PLifted PTxInInfo = Plutus.TxInInfo
+
+-- | @since 2.0.0
+deriving via
+  (DerivePConstantViaData Plutus.TxInInfo PTxInInfo)
+  instance
+    PConstantDecl Plutus.TxInInfo
+
+-- | @since 2.0.0
+data POutputDatum (s :: S)
+  = PNoOutputDatum (Term s (PDataRecord '[]))
+  | POutputDatumHash (Term s (PDataRecord '["datumHash" ':= PDatumHash]))
+  | -- | Inline datum as per
```

Comment:
Don't need this documentation about CIP. Rather, it needs to explain _**what inline-datum is**_. 

This is not under the scope of this PR.